python/pfs/pipe2d/generateCommands.py

```diff
@@ -889,12 +881,40 @@ class CalibBlock:
     # if it were to be autogenerated.
     calibTypes = ["bias", "dark", "flat", "bootstrap", "fiberTrace", "detectorMap"]
 
-    def __init__(self, blockName: str, yamlBlock: Mapping[str, Any]):
-        self.name = blockName
-        self.sources = {}
+    def __init__(self, name: str, sources: Mapping[str, CalibSource]):
+        self.name = name
+        self.sources = dict(sources)
 
-        for typeName, block in yamlBlock.items():
-            self.sources[typeName] = CalibSource.getSubclass(typeName).fromYaml(block)
+    @classmethod
+    def fromYaml(cls, yamlBlock: Mapping[str, Any]) -> "CalibBlock":
+        """Construct ``CalibBlock`` from a YAML block.
+
+        Parameters
+        ----------
+        yamlBlock : `Mapping[str, Any]`
+            A block of a YAML data structure:
+            This has to be a mapping from calib types to CalibSources
+            with a special key "name", which defines the name of this block.
+            For example:
+                name: "myblock"
+                bias:
+                    id: "visit=0..4"
+                    config: ["key=value", "key=value"]
+                dark:
+                    id: ["field=DARK", "dateObs=2000-01-01"]
+                    configfile: "config.py"
+
+        Returns
+        -------
+        calibBlock : `CalibBlock`
+        """
+        name = yamlBlock["name"]
+        sources = {
+            typeName: CalibSource.getSubclass(typeName).fromYaml(block)
+            for typeName, block in yamlBlock.items()
+            if typeName != "name"
```

Comment:
I had to read this a couple of times (I was wondering where the `:` was at the end of the `for` line), so I wonder if this is the best way to break the lines.